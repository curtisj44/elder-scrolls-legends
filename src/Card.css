.card {
  padding: 3% 5%;
  background: var(--corbeau);
  text-align: center;
}

/* TODO: use BEM */

.card dt,
.card dd {
  padding: .6rem 8%;
}

.card dt {
  color: var(--pumpkin-pie);
  font: .7rem var(--serif);
  letter-spacing: .07em;
  padding-bottom: .2rem;
  text-transform: uppercase;
}

.card dd {
  font-size: .9rem;
  padding-top: 0;
}

/* Images are either 409x663 or 350x580 */
.card img {
  min-height: 366px;
  width: 350px;
}
