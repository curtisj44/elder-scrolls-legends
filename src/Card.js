// Dependencies
// ------------

// Libraries
import React from 'react';
// Styles
import './Card.css';

// Internal
// --------

// TODO: add propTypes

const Card = ({ imageUrl, name, set, text, type }) => {
  return (
    <li className="card">
      {/*
        1. Using `aria-hidden="true"` since the content of this image is duplicated in text

        2. Image sizes are either 409x663 or 350x580. Using the smaller size in the `width`
           and `height` attributes to improve performance with use of `loading` attribute
      */}
      <img
        alt=""
        aria-hidden="true"
        height="580"
        loading="lazy"
        src={ imageUrl }
        width="350"
      />

      <dl>
        <dt>Name</dt>
        <dd>{ name }</dd>

        { text &&
          <>
            <dt>Text</dt>
            <dd>{ text }</dd>
          </>
        }

        <dt>Set Name</dt>
        <dd>{ set.name }</dd>

        <dt>Type</dt>
        <dd>{ type }</dd>
      </dl>
    </li>
  );
};

// Exports
// -------

export default Card;
