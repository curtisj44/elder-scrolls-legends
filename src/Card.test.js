// Dependencies
// ------------

// Libraries
import React from 'react';
import renderer from 'react-test-renderer';
// Components
import Card from './Card';

// Setup
// -----

const defaultProps = {
  imageUrl: 'imageUrl-mock',
  name: 'name-mock',
  set: {
    name: 'set-mock'
  },
  type: 'type-mock'
};

let component;

const getComponent = (props) => (
  renderer.create(
    <Card { ...{ ...defaultProps, ...props } } />
  )
);

// Tests
// -----

describe('Card', () => {
  describe('when `text` prop is not defined', () => {
    it('renders without Text content', () => {
      component = getComponent();
      expect(component).toMatchSnapshot();
    });
  });

  describe('when `text` prop is defined', () => {
    it('renders with Text content', () => {
      component = getComponent({ text: 'text-mock' });
      expect(component).toMatchSnapshot();
    });
  });
});
