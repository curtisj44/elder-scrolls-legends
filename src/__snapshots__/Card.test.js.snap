// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card when \`text\` prop is defined renders with Text content 1`] = `
<li
  className="card"
>
  <img
    alt=""
    aria-hidden="true"
    height="580"
    loading="lazy"
    src="imageUrl-mock"
    width="350"
  />
  <dl>
    <dt>
      Name
    </dt>
    <dd>
      name-mock
    </dd>
    <dt>
      Text
    </dt>
    <dd>
      text-mock
    </dd>
    <dt>
      Set Name
    </dt>
    <dd />
    <dt>
      Type
    </dt>
    <dd>
      type-mock
    </dd>
  </dl>
</li>
`;

exports[`Card when \`text\` prop is not defined renders without Text content 1`] = `
<li
  className="card"
>
  <img
    alt=""
    aria-hidden="true"
    height="580"
    loading="lazy"
    src="imageUrl-mock"
    width="350"
  />
  <dl>
    <dt>
      Name
    </dt>
    <dd>
      name-mock
    </dd>
    <dt>
      Set Name
    </dt>
    <dd />
    <dt>
      Type
    </dt>
    <dd>
      type-mock
    </dd>
  </dl>
</li>
`;
